File: Top_Jobs_ResumeService/src/main/java/com/cgi/topjobs/api/resume/services/ResumeService.java
package com.cgi.topjobs.api.resume.services;

import java.util.List;

import com.cgi.topjobs.api.resume.dao.model.EmployeeShortListedCandidates;
import com.cgi.topjobs.api.resume.dto.model.EmployeesShortlisted;
import com.cgi.topjobs.api.resume.dto.model.Resume;

public interface ResumeService {
	/**
	 * AbstractMethod to create a resume
	 */
	Resume createResume(Resume resume);

	/**
	 * AbstractMethod to get all resumes
	 */
	List<Resume> getAllResumes();

	/**
	 * AbstractMethod to get resumes by jobid
	 */
	List<EmployeeShortListedCandidates> getResumesByJobId(Integer id);

	/**
	 * AbstractMethod to get resumes by resume id
	 */
	Resume getResumesByResumeId(String id);

	/**
	 * AbstractMethod to shortlist resume
	 */
	public EmployeesShortlisted shortListResume(EmployeesShortlisted shortListedResume);

	List<EmployeeShortListedCandidates> getShortlistedPeople();

}

File: angular-Shared/resume.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ResumeService {
headers;
  constructor(private http:HttpClient) {
    this.headers = new HttpHeaders({'Content-Type':'application/json'});
   }
   createResume(resume) : Observable<any> {
    return this.http.post('http://10.0.0.198:8011/resume-ws/api/v1/resume/create', resume, {headers: this.headers});
 }
 getAllResumes() : Observable<any> {
  return this.http.get('http://10.0.0.198:8011/resume-ws/api/v1/resumes');
}
getResumesByJobId(id) :  Observable<any> {
  return this.http.get('http://10.0.0.198:8011/resume-ws/api/v1/resume/search/{JobId}' + id);

}
getResumesByResumeId(id) :  Observable<any> {
  return this.http.get('http://10.0.0.198:8011/resume-ws/api/v1/resume/search/resumes/{ResumeId}' + id);
  


}
shortListResume(shortListedResume){
  return this.http.post('http://10.0.0.198:8011/resume-ws/api/v1/resume/shortlist',shortListedResume , {headers: this.headers});
}

getShortList() {
  return this.http.get('http://10.0.0.198:8011/resume-ws/api/v1/resume/shortlist/resumes');
}

}